docs: add page for the new ALTER SYSTEM command (DOCS-14391) (#9149)
confluentinc/ksql: docs/developer-guide/ksqldb-reference/alter-system.md

---
layout: page
title: ALTER SYSTEM
tagline:  ksqlDB ALTER SYSTEM statement
description: Syntax for the ALTER SYSTEM statement in ksqlDB
keywords: ksqlDB, system, config, properties
---

ALTER SYSTEM
==========

Synopsis
--------

```sql
ALTER SYSTEM '<config-name>'='<value>';
```

Description
-----------

> You can use this command to set and modify the value of system-level configs. These

Reviewer: ```suggestion
Sets and modifies the value of system-level configs. These
```

> are properties that are applied to all queries in the cluster, and using `ALTER SYSTEM` 

Reviewer: ```suggestion
are properties that are applied to all queries in the cluster. Using `ALTER SYSTEM` 
```

> will result in the new value immediately being applied across the cluster.

Reviewer: ```suggestion
results in the new value being applied immediately across the cluster.
```

`ALTER SYSTEM` is only available from the ksql command line and only when connected to a

> ksql cluster in CCloud.

Reviewer: ```suggestion
ksqlDB cluster in {{ site.ccloud }}.
```

Example
-------

> The following statement sets the streams property `auto.offset.reset` to pick up from

Reviewer: ```suggestion
The following statement sets the streams property `auto.offset.reset` to begin at the
```

> `earliest` in the input topics when no committed offsets are found.

Reviewer: ```suggestion
`earliest` record in the input topics when no committed offsets are found.
```
